pkgs/development/tools/godot/mono/make-deps.nix

```diff
@@ -0,0 +1,71 @@
+{ godot-mono
+, lib
+, nuget-to-nix
+, writeScriptBin
+, path
+, stdenv
+}:
+
+# This can be used to recreate deps.nix. This entails doing things that are not permitted
+# with a sandboxed Nix build, so instead of building the deps.nix directly, it builds a
+# script that can be run to generate it.  Thus, to use this derivation, simply do:
```

Comment:
I'd rather leave the more elaborate explanation in place.  Not everyone is aware of internet connectivity being disallowed within the nix build sandbox.  But even if you assume they are, they're not necessarily aware that running the `buildPhase` without having run the `configureNugetStep` results in internet access.  I could reword the comments to be clearer about *that* rather than saying "doing things that are not permitted within a sandboxed Nix build".  The comments also explain that this derivation doesn't build deps.nix, but instead builds a script that builds deps.nix.

Reply:
> Not everyone is aware of internet connectivity being disallowed within the nix build sandbox.

I hope everybody is. This is a vital part of how nix build works and we don't need to educate people so deep in code. This should be part of an entry level documentation that gives an overview over everything.

> But even if you assume they are, they're not necessarily aware that running the `buildPhase` without having run the `configureNugetStep` results in internet access

I am not sure how someone could do this by accident without modifying the derivation first.